web-services/security/src/main/java/datawave/security/user/UserOperationsBean.java

```
@@ -137,10 +182,22 @@ public AuthorizationsListBase listEffectiveAuthorizations() {
     @Produces({"application/xml", "text/xml", "application/json", "text/yaml", "text/x-yaml", "application/x-yaml", "application/x-protobuf",
             "application/x-protostuff"})
     @PermitAll
-    public GenericResponse<String> flushCachedCredentials() {
+    public GenericResponse<String> flushCachedCredentials(@DefaultValue("true") @QueryParam("includeRemoteServices") boolean includeRemoteServices) {
         GenericResponse<String> response = new GenericResponse<>();
         Principal callerPrincipal = context.getCallerPrincipal();
         log.info("Flushing credentials for " + callerPrincipal + " from the cache.");
+        
+        // if we have any remote services configured, then flush those credentials as well
+        if (includeRemoteServices && remoteServices != null && !remoteServices.isEmpty()) {
+            for (RemoteUserOperations remote : remoteServices) {
+                try {
+                    remote.flushCachedCredentials(callerPrincipal);
+                } catch (Exception e) {
+                    log.error("Failed to flush user from remote user service", e);
```

Consider how we might alert the caller that the remote operation failed. It would not be fun to be the person calling the cache flush to resolve an issue and having it silently fail on the remote with no indication other than the logs.

---

Added a message attribute to the response which now get populated.